src/main/java/com/squareup/tools/maven/resolution/HttpArtifactFetcher.kt

```diff
@@ -38,18 +39,35 @@ import org.apache.maven.model.Repository
  * This class currently does not support SNAPSHOT artifacts.
  * TODO: Handle snapshots
  */
+
 class HttpArtifactFetcher(
-  cacheDir: Path,
-  private val client: OkHttpClient = OkHttpClient()
```

Comment:
turns this into a private val client takes a url and returns a client.

Move logic from fetchFile into a function and takes a URL.

Call the lambda inside fetchFile as a property -> extracted logic (pass function reference).  

Move code into ProxyHelper.  The default value of the lambda is the function reference

Reply:
Turns it into a private val which is a LAMBDA which take a url and returns a client, yes. 

Otherwise, yes, this sounds exactly like what we talked about.

This moves configuration of the client outside this class, but leaves it flexible to return different clients for different URLs. It also means we can optimize without changing the APIs later.

Reply:
So - let's just get this in, and I can roll a refactor subsequently.